### ISSUE TITLE
shadow variable: 'undefined'

### ISSUE THREAD

**Issue by alitaheri** [palantir/tslint#828]
to reproduce:

lint this line of code: 
```js
const a = ([b, c]) => ([d, e]) => { };
```

with `no-shadowed-variable` set to true.

I believe nested array destructing of arguments breaks this.

**Comment by IllusionMH** [comment]
I can't reproduce this issue with `master` branch (TSLint v3.14.0, TypeScript 1.8.10).

```js
const a = ([b, c]) => ([d, e]) => { };
```
Produces no errors when I run it from CLI (or add it to `test/rules/no-shadowed-variable/test.ts.lint`)

However if I replace `e` with `a` than I see valid error
```js
const a = ([b, c]) => ([d, a]) => { };
                           ~           [Shadowed variable: 'a']
```

**Comment by adidahiya** [comment]
this bug report is very old. if you encounter this issue again, please comment to reopen.

**Comment by alitaheri** [comment]
I tested the rule with my project. The issue is resolved. Thanks a lot guys, keep rocking :grin:  :+1:
